src/shared/internal-calendar/internal-calendar.tsx

```diff
@@ -450,56 +455,63 @@ export const Component = (
-                        between={between}
-                        isNewSelection={isNewSelection}
-                        onSelect={handleDateSelect}
-                        onHover={handleHover}
-                    />
-                    <OptionsOverlay $visible={currentView !== "default"}>
-                        {renderOptionsOverlay()}
-                    </OptionsOverlay>
-                </ToggleZone>
-                {renderCancelDoneButton()}
-            </ContentBody>
-            <SideArrowButton
-                $direction="right"
-                $type={type}
-                onClick={handleRightArrowClick}
-            >
-                <ArrowRight />
-            </SideArrowButton>
-        </Container>
-    );
+    const renderContent = () => {
+        return (
+            <Container ref={resizeDetector.ref} tabIndex={-1} $type={type}>
+                {type === "standalone" && (
+                    <SideArrowButton
+                        $direction="left"
+                        onClick={handleLeftArrowClick}
+                    >
+                        <ArrowLeft />
+                    </SideArrowButton>
+                )}
+                <ContentBody>
+                    {renderHeader()}
+                    <ToggleZone>
+                        <InternalCalendarDay
+                            type={type}
+                            calendarDate={calendarDate}
+                            currentFocus={currentFocus}
+                            disabledDates={disabledDates}
+                            selectedStartDate={selectedStartDate}
+                            selectedEndDate={selectedEndDate}
+                            variant={variant}
+                            between={between}
+                            isNewSelection={isNewSelection}
+                            onSelect={handleDateSelect}
+                            onHover={handleHover}
+                        />
+                        <OptionsOverlay $visible={currentView !== "default"}>
+                            {renderOptionsOverlay()}
+                        </OptionsOverlay>
+                    </ToggleZone>
+                    {renderCancelDoneButton()}
+                </ContentBody>
+                {type === "standalone" && (
+                    <SideArrowButton
+                        $direction="right"
+                        onClick={handleRightArrowClick}
+                    >
+                        <ArrowRight />
+                    </SideArrowButton>
+                )}
+            </Container>
+        );
+    };
+
+    if (type === "input") {
+        // React spring animation configuration
+        const styles = useSpring({
+            height: isOpen
+                ? resizeDetector.height + 64 // include vertical padding
+                : 0,
+        });
+
+        return <AnimatedDiv style={styles}>{renderContent()}</AnimatedDiv>;
+    } else {
+        return <>{renderContent()}</>;
```

Comment:
a wrapper is still needed for the standalone calendar to set className prop and/or constrain the width. it can be added to `calendar.tsx`